#!/usr/bin/env python
PACKAGE = "swanson_controls"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("target_angle", double_t, 0, "Target Angle",    90.0, -180.0, 180.0)
gen.add("kp",           double_t, 0, "P gain",          10.0, 0.0, 20.0)
gen.add("ki",           double_t, 0, "I Gain",          3.0, 0.0, 10.0)
gen.add("kd",           double_t, 0, "D Gain",          0.05, 0.0, 10.0)
gen.add("max_angle",    double_t,   0, "Max Angle",     180.0, -180.0, 180.0)
gen.add("null_cmd",    int_t,   0, "PWM Pulse associated with stop", 1590, 800, 2300)

# size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                        gen.const("Medium",     int_t, 1, "A medium constant"),
#                        gen.const("Large",      int_t, 2, "A large constant"),
#                        gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
#                      "An enum to set size")
#
# gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "swanson_controls", "Gimbal"))
